plugin/logging.go

```diff
@@ -0,0 +1,64 @@
+package plugin
+
+import (
+	"context"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend/log"
+	"go.opentelemetry.io/otel/trace"
+
+	"github.com/grafana/grafana-app-sdk/logging"
+)
+
+// NewLogger returns a new PluginLogger that wraps the provided log.Logger
+func NewLogger(l log.Logger) *PluginLogger {
+	return &PluginLogger{
+		Logger: l,
+	}
+}
+
+// PluginLogger wraps a plugin-sdk-go log.Logger with the context methods needed to implement logging.Logger,
+// and automatically adds the traceID from the context to the log.Logger's args when DebugContext,
+// InfoContext, WarnContext, or ErrorContext are called.
+// nolint:revive
+type PluginLogger struct {
+	log.Logger
+	ctx context.Context
+}
+
+// Debug adds the traceID field to the underlying log.Logger from the context embedded with WithContect, then calls Debug with the provided msg and args
+func (p *PluginLogger) Debug(msg string, args ...any) {
+	p.Logger.With(logging.TraceIDKey, trace.SpanContextFromContext(p.ctx).TraceID()).Debug(msg, args...)
```

Comment:
Should we check the presence of span in the context here? Otherwise I think we might get some logs with empty traceID fields.

Reply:
Yup, makes sense, I was trying to maximize for efficiency, but probably not the best place to do so.